presto-main/src/main/java/com/facebook/presto/server/CoordinatorModule.java

```diff
@@ -268,13 +271,20 @@ protected void setup(Binder binder)
         binder.bind(RemoteTaskStats.class).in(Scopes.SINGLETON);
         newExporter(binder).export(RemoteTaskStats.class).withGeneratedName();
 
-        httpClientBinder(binder).bindHttpClient("scheduler", ForScheduler.class)
-                .withTracing()
-                .withFilter(GenerateTraceTokenRequestFilter.class)
-                .withConfigDefaults(config -> {
-                    config.setRequestTimeout(new Duration(10, SECONDS));
-                    config.setMaxConnectionsPerServer(250);
-                });
+        ReactorNettyHttp2ClientConfig reactorNettyHttp2ClientConfig = buildConfigObject(ReactorNettyHttp2ClientConfig.class);
+        if (reactorNettyHttp2ClientConfig.isReactorNettyHttp2ClientEnabled()) {
+            binder.bind(ReactorNettyHttp2Client.class).in(Scopes.SINGLETON);
+            binder.bind(HttpClient.class).annotatedWith(ForScheduler.class).to(ReactorNettyHttp2Client.class);
+        }
+        else {
+            httpClientBinder(binder).bindHttpClient("scheduler", ForScheduler.class)
+                    .withTracing()
+                    .withFilter(GenerateTraceTokenRequestFilter.class)
+                    .withConfigDefaults(config -> {
+                        config.setRequestTimeout(new Duration(10, SECONDS));
```

Comment:
We probably need to configure request and connect timeouts for the Reactor client as well